src/wp-includes/block-template-utils.php

```diff
@@ -383,12 +388,18 @@ function _get_block_templates_files( $template_type, $query = array() ) {
 
 			if ( 'wp_template' === $template_type ) {
 				$candidate = _add_block_template_info( $new_template_item );
+				$is_custom = ! isset( $default_template_types[ $candidate['slug'] ] );
+
 				if (
 					! $post_type ||
 					( $post_type && isset( $candidate['postTypes'] ) && in_array( $post_type, $candidate['postTypes'], true ) )
 				) {
 					$template_files[ $template_slug ] = $candidate;
 				}
+
+				if ( $post_type && ! isset( $candidate['postTypes'] ) && $is_custom ) {
+					$template_files[ $template_slug ] = $candidate;
+				}
```

Comment:
Can we combine two conditions with OR?

Reply:
I think it's easier to parse this way. I see no reason to sacrifice readability for an extra few lines.